src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -440,10 +440,6 @@ void ImageReplayer<I>::handle_prepare_local_image(int r) {
   } else if (r < 0) {
     on_start_fail(r, "error preparing local image for replay");
     return;
-  } else if (m_local_image_tag_owner == librbd::Journal<>::LOCAL_MIRROR_UUID) {
-    dout(5) << "local image is primary" << dendl;
-    on_start_fail(0, "local image is primary");
-    return;
```

Comment:
@dillaman Running stress test locally I observed a false failure, when an ImageReplayer failed on start with the status "state=up+error, description=remote image does not exist", while it was expected to be "state=up+stopped, description=local image is primary".

The sequence was following:
1) a mirrored image (primary) is created in cluster2;
2) the local pool watcher for cluster2 mirror daemon gets notification about the new image
3) cluster2 mirror daemon starts the image replayer
4) start fails in prepare_remote_image because the image does not exist in the remote cluster yet.

If my explanation is not clear or you would like to see details I can provide logs.

Reply:
@trociny Thanks -- good catch. Update pushed to ignore the errors from prepare remote if the local image is primary.